src/libsyntax/ext/base.rs

```diff
@@ -73,6 +73,108 @@ impl<F> ItemModifier for F
     }
 }
 
+#[derive(Show,Clone)]
+pub enum Annotatable {
+    Item(P<ast::Item>),
+    TraitItem(ast::TraitItem),
+    ImplItem(ast::ImplItem),
+}
+
+impl Annotatable {
+    pub fn attrs(&self) -> Vec<ast::Attribute> {
```

Comment:
It works! (Pretty impressed rustc is able to figure out the lifetimes with that)